osidb/models.py

```diff
@@ -1788,6 +1789,55 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, _type, url, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(flaw=flaw, type=_type, url=url)
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+
+        except ObjectDoesNotExist:
+            return FlawReference(flaw=flaw, type=_type, url=url, **extra_fields)
+
+
+class FlawReference(TrackingMixin, ACLMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """allowable references"""
+
+        VULN_RESPONSE = "VULN_RESPONSE"
+        EXTERNAL = "EXTERNAL"
+
+    # internal primary key
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    type = models.CharField(
+        choices=FlawReferenceType.choices,
+        default=FlawReferenceType.EXTERNAL,
+        max_length=50,
+    )
+
+    url = models.CharField(max_length=200)
+
+    description = models.TextField(blank=True)
+
+    # one flaw can have many references
+    flaw = models.ForeignKey(Flaw, on_delete=models.CASCADE, related_name="references")
+
+    objects = FlawReferenceManager()
+
+    class Meta:
+        """define meta"""
+
+        unique_together = ["flaw", "type", "url"]
```

Comment:
does it make sense to have the same URL twice, once per type? if not, I'd just have it be unique on flaw and URL

Reply:
Probably it does not. Why should we have the same URL as two different types... Makes sense to me to simplify.